Unable to use LOG_BACKEND_DEFINE macro from log_backend.h using C++

**Describe the bug**
When including log_backend.h from a C++ application, the following error is produced by the C++ compiler:

`zephyr/include/logging/log_backend.h:81:2: error: designator order for field 'log_backend_control_block::id' does not match declaration order in 'log_backend_control_block'`

**To Reproduce**
Steps to reproduce the behavior:
1. Add the following code to a C++ source file

`
#include <logging/log_backend.h>
const struct log_backend_api my_backend_api = {};
LOG_BACKEND_DEFINE(my_backend, my_backend_api, true);
`
**Expected behavior**
A compilation error is not produced.

**Impact**
Annoyance.  A workaround is to #undef the macro and redefine it with correct order of arguments.

**Fix**
The fix is to reverse the order of the .id and .active initialization lines in the LOG_BACKEND_DEFINE macro.

**Environment (please complete the following information):**
 - OS: Fedora
 - Toolchain Zephyr riscv SDK
 - Commit SHA 4fea711f5e9ab9e20d26fb570edd302f56f51c4a

--- a/include/logging/log_backend.h
+++ b/include/logging/log_backend.h
@@ -76,8 +76,8 @@ extern const struct log_backend __log_backends_end[0];
 #define LOG_BACKEND_DEFINE(_name, _api, _autostart)			       \
 	static struct log_backend_control_block UTIL_CAT(backend_cb_, _name) = \
 	{								       \
-		.active = false,					       \
 		.id = 0,						       \
+		.active = false,					       \
 	};								       \
 	static const Z_STRUCT_SECTION_ITERABLE(log_backend, _name) =	       \
 	{								       \

